### PR TITLE
[plotly-js] Added missing parameters

### PR DESCRIPTION
**Pull request by afl-h**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## plotly.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/plotly.js@latest/))
was missing the following properties:
1. version
1. Fx
1. Snapshot
1. PlotSchema
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7868de05b784a1252ebdf5b0d57a8c6960a555e8
</p>

File: types/plotly.js/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for plotly.js 2.12
+// Type definitions for plotly.js 2.24
```
Comment (brammitch) [inline types/plotly.js/index.d.ts:1]: I do not see any reason to update the version to 2.24. Looking at [plotly.js/src/traces/bar/attributes.js](https://github.com/plotly/plotly.js/blob/96c28159c06a3af3fe551175db89599894efbf82/src/traces/bar/attributes.js#L74), git blame on the properties of `textposition`, `insidetextanchor`, and `textangle` shows all of them to be at least 4 years old.

Comment (brammitch) [reply types/plotly.js/index.d.ts:1]: Recommend reverting this change.

File: types/plotly.js/index.d.ts [hunk 2]
```diff
@@ -1300,20 +1300,14 @@ export interface PlotData {
         | 'percent'
         | 'none';
     textposition:
```
Comment (brammitch) [inline types/plotly.js/index.d.ts:1302]: It looks like you narrowed the scope of the `textposition` to just the [Bar trace](https://github.com/plotly/plotly.js/blob/96c28159c06a3af3fe551175db89599894efbf82/src/traces/bar/attributes.js#L74), but other traces, like [Scatter](https://github.com/plotly/plotly.js/blob/96c28159c06a3af3fe551175db89599894efbf82/src/traces/scatter/attributes.js#L620) still use other values such as `top left`, `bottom right`, etc.

Recommend reverting this change.

**Review by brammitch — CHANGES_REQUESTED** [review]
The addition of `textangle`, `insidetextanchor`, and `constraintext` look good. Other changes should be removed.

File: types/plotly.js/index.d.ts (hunk 2 above)
Comment (afl-h) [reply types/plotly.js/index.d.ts]: Reverted the changes

**Review by peterblazejewicz — APPROVED** [review]
@afl-h thanks!

**Comment by afl-h** [comment]
Ready to merge

**Pull request merged**